spec-tests/src/test/java/org/candlepin/spec/pools/RefreshPoolsTest.java

```diff
@@ -0,0 +1,877 @@
+        }
+        else {
+            assertNull(status.getResultData());
+        }
+
+        OwnerDTO owner = adminClient.owners().getOwner(expectedKey);
+        assertNotNull(owner);
+        assertEquals(expectedKey, owner.getDisplayName());
+        assertEquals(Owners.SCA_ACCESS_MODE, owner.getContentAccessMode());
+    }
+
+    @Test
+    @OnlyInHosted
+    void shouldCreateTheCorrectNumberOfPools() {
+        int expectedSubscriptionCount = 6;
+        Set<String> expectedSubIds = new HashSet<>();
+        for (int i = 0; i < expectedSubscriptionCount; i++) {
+            ProductDTO prod = adminClient.hosted().createProduct(Products.random());
+            SubscriptionDTO sub = adminClient.hosted().createSubscription(Subscriptions.random(owner, prod));
+            expectedSubIds.add(sub.getId());
+        }
+
+        refreshPools(adminClient, ownerKey);
+
+        List<PoolDTO> pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(expectedSubscriptionCount, pools.size());
+        for (PoolDTO pool : pools) {
+            assertNotNull(pool.getSubscriptionId());
+            assertTrue(expectedSubIds.contains(pool.getSubscriptionId()));
+        }
+    }
+
+    @Test
+    @OnlyInHosted
+    void shouldDetectChangesInProviderProducts() {
+        ProductDTO prod1 = adminClient.hosted().createProduct(Products.random());
+        ProductDTO prod2 = adminClient.hosted().createProduct(Products.random());
+        ProductDTO providerProduct = Products.random();
+        providerProduct.setProvidedProducts(Set.of(prod1, prod2));
+        providerProduct = adminClient.hosted().createProduct(providerProduct);
+        adminClient.hosted().createSubscription(Subscriptions.random(owner, providerProduct));
+
+        refreshPools(adminClient, ownerKey);
+        List<PoolDTO> pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+
+        PoolDTO actualPool = pools.get(0);
+        Set<String> expectedProvProdIds = Set.of(prod1.getId(), prod2.getId());
+        assertEquals(expectedProvProdIds.size(), actualPool.getProvidedProducts().size());
+        for (ProvidedProductDTO provProd : actualPool.getProvidedProducts()) {
+            assertTrue(expectedProvProdIds.contains(provProd.getProductId()));
+        }
+
+        // Remove the old provided products and add a new one...
+        ProductDTO prod3 = adminClient.hosted().createProduct(Products.random());
+        providerProduct.setProvidedProducts(Set.of(prod3));
+        providerProduct = adminClient.hosted().updateProduct(providerProduct.getId(), providerProduct);
+
+        refreshPools(adminClient, ownerKey);
+        pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+        PoolDTO pool = pools.get(0);
+        assertEquals(1, pool.getProvidedProducts().size());
+        assertEquals(prod3.getId(), pool.getProvidedProducts().stream()
```

Comment:
This stream collector seems like a really complicated way of doing:

`<collection>.iterator().next()`

Given that we're pulling the provided products collection twice, it may even make sense to do that above and then use assertJ tests on it:

```
Collection<ProductDTO> providedProducts = pool.getProvidedProducts();
assertThat(providedProducts)
  .isNotNull()
  .hasSize(1)
  .singleElement()
  .returns(prod3.getId(), ProductDTO::getId);
```

Or you could do the whole block as one long chain, as mentioned in another comment somewhere in this review.